### PR TITLE
ADEN-2400 Refactor

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 31 issues:
* MAJOR 1 major
* MINOR 30 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Define a constant instead of duplicating this literal "MH3U:_Monsters" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L102) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/Krux/Page_1" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L438) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "wka.hub/_life_hub//hub" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L85) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "src/test/resources/adsResources/wikia_skin_right.png" 9 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L213) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "CORP_TOP_LEADERBOARD" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "adtest" 38 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L33) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "DirectGpt; RemnantGpt; Liftium" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L401) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "yugioh" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L44) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "%D0%9A%D0%B2%D0%B5%D1%81%D1%82%D1%8B_%28Skyrim%29" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L98) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "FFFFFF" 8 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L215) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "src/test/resources/adsResources/wikia_skin_left.png" 9 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L212) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "adtest-pad" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L500) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "elderscrolls" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "TOP_RIGHT_BOXAD" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L177) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "naruto" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L45) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "monsterhunter" 5 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L102) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "ru.elderscrolls" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L98) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "gameshub" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "zh.pad" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L53) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "wikia" 7 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/Interstitial" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L511) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/ProvidersChain" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L400) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "126608052" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L165) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/Interstitial/NotScalable" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L518) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "TOP_LEADERBOARD" 10 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L165) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "Wikia" 4 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "HUB_TOP_LEADERBOARD" 8 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "leagueoflegends" 3 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L46) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "AAAAAA" 9 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L214) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "adtest-pluto" 8 times.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L104) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Return an empty collection instead of null.](https://github.com/Wikia/selenium-tests/blob/8719d777b97ee9700690372cca501546d605ad29/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L551) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsGptPageParamOasis.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.wikia.webdriver.common.driverprovider.UseUnstablePageLoadStrategy;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+
+import org.testng.annotations.Test;
+
+/**
+ * @ownership AdEng
+ */
+public class TestAdsGptPageParamOasis extends TemplateNoFirstLoad {
```
Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsGptPageParamOasis.java]: Do we really only use it for Oasis?

Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsGptPageParamOasis.java]: https://github.com/Wikia/selenium-tests/blob/aden-2400/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsGptPageParamOasis.java#L23

As you can see this method is alone here.

Comment (nandy-andy) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsGptPageParamOasis.java]: Yeah. As we discussed: I don't see any reason why we couldn't re-use this test for Mercury as well. But let's leave it as-is and rename it once we really use it for Mercury too ;)

**Comment by nandy-andy** [comment]
LGTM :+1:

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsKruxObject.java [hunk 2]
```diff
@@ -55,16 +52,20 @@ public void verifyKruxControlTag(String kruxSiteId) {
...
-  public void verifyKruxUserParam() {
+  public void verifyKruxUserParam(String slotName) {
     String script = "return localStorage.kxuser;";
     waitForKrux();
     String user1 = (String) ((JavascriptExecutor) driver).executeScript(script);
     refreshPage();
     waitForKrux();
     String user2 = (String) ((JavascriptExecutor) driver).executeScript(script);
+    String gptPageParams = getGptPageParams(slotName);
     // TODO: figure out why we get krux user id in GPT calls from localStorage.kxuser in current PV OR from previous PV
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsKruxObject.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 34 issues:
* MAJOR 1 major
* MINOR 32 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/core/Assertion.java#L106) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Define a constant instead of duplicating this literal "assertion " 5 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/core/Assertion.java#L30) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "gameshub" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "monsterhunter" 5 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L102) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/Interstitial" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L511) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "elderscrolls" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "wka.hub/_life_hub//hub" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L85) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "TOP_LEADERBOARD" 10 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L165) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "naruto" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L45) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "zh.pad" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L53) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "AAAAAA" 9 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L214) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "ru.elderscrolls" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L98) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "adtest-pad" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L500) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "adtest-pluto" 8 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L104) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "DirectGpt; RemnantGpt; Liftium" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L401) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "FFFFFF" 8 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L215) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/ProvidersChain" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L400) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "leagueoflegends" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L46) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/Krux/Page_1" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L438) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "src/test/resources/adsResources/wikia_skin_right.png" 9 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L213) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "126608052" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L165) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "HUB_TOP_LEADERBOARD" 8 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "yugioh" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L44) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "Wikia" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "wikia" 7 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "CORP_TOP_LEADERBOARD" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "src/test/resources/adsResources/wikia_skin_left.png" 9 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L212) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "adtest" 38 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L33) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "%D0%9A%D0%B2%D0%B5%D1%81%D1%82%D1%8B_%28Skyrim%29" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L98) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "TOP_RIGHT_BOXAD" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L177) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "SyntheticTests/Interstitial/NotScalable" 4 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L518) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MINOR [Define a constant instead of duplicating this literal "MH3U:_Monsters" 3 times.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/common/dataprovider/ads/AdsDataProvider.java#L102) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [Return an empty collection instead of null.](https://github.com/Wikia/selenium-tests/blob/6a284d94ba6e3715278d8614b102f3984f1f4ba9/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L551) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)

**Pull request merged**